log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java

```diff
@@ -256,7 +254,7 @@ protected void append(LoggingEvent event) {
             try {
                 response.get();
             } catch (InterruptedException | ExecutionException ex) {
-                throw new RuntimeException(ex);
+                throw new KafkaException(ex);
```

Comment:
The error was returned by the KafkaProducer,    so I've wrapped it in KafkaException.